### ISSUE TITLE
Decorators break code with es2019 target

### ISSUE THREAD

**Issue by ryuhhnn** [swc-project/swc#846]
**Describe the bug**

If you have a class that extends a base class such as the following:

```js
class SomeClass {
  someMethod() {}
}

class OtherClass extends SomeClass {
  anotherMethod() {
    super.someMethod()
  }
}
```

You will get the following error: `super is only allowed in object methods and classes` because the compiler will output `OtherClass` as a returned object thus inhibiting the ability to use the `this` keyword. 

This is similar to an issue that Babel was experiencing a few years ago: https://github.com/babel/babel/issues/2775

**Config**
```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "decorators": true,
      "decoratorsBeforeExport": true,
      "classProperty": true,
      "nullishCoalescing": true,
      "optionalChaining": true
    },
    "target": "es2019"
  }
}
```

**Expected behavior**
A class should be able to call `super.*` on its parent class.

**Version**
```
@swc/cli@^0.1.24
@swc/core@^1.2.2
```

**Comment by kdy1** [comment]
>     "target": "es2019"

This part is problematic. Because you told swc that you target es2019, swc does **not** transcompile classes, so it's just emitted as-is.
I have added some tests anyway.

**Comment by ryuhhnn** [comment]
Not sure why this issue was closed, I provided a valid case where this breaks and it wasn't fixed. I understand that swc does not transpile classes when the target is es2019, but if that is the case, then the decorator transpilation needs to be modified to not return an object and instead return a class. Is this something I can help with? I really want to start using swc, but this is a critical bug preventing us from adopting.

**Comment by kdy1** [comment]
I missed the decorator part as it was not in example. I'm Sorry for it.

**Comment by Brooooooklyn** [comment]
Decorator should always be transpiled. Now it would only be transpiled under target <= `es2015`

**Comment by kdy1** [comment]
@Brooooooklyn It's transpiled iff syntax.decorators is true, but it seems like transpiled code crashes for some reason

**Comment by tvarsis** [comment]
There seems to be a lot of mentioned of this ticket from different milestones, but it does not seems resolved yet. Is there any update on this issue which makes our project fail to transpile?

**Comment by kdy1** [comment]
@tvarsis 

I added this to the milestones because I wanted to fix it.
But the project is maintained by a one-man team and I couldn't fix it because I don't have time.
I have lots of tasks to do.

**Comment by kdy1** [comment]
I did some investigation on this.
The decorator pass converts a class into functions and `classes` pass fails to detect super calls because it's not in a class.


The fix would be handling super accesses in `decorator` pass, just like `class_properties` pass. A simple way would be making `swc_ecma_transforms_proposal` depend on the `swc_ecma_transforms_compat`, but it makes another problem.

Deno depends on `swc_ecma_transforms_proposal` but not on `swc_ecma_transforms_compat`, and I don't want to increase the build time of deno. I tried moving the super access handler to `swc_ecma_utils`, but it didn't work because the super handler depends on helpers, which is defined in `swc_ecma_transforms_base`.


---

So there are two ways.

 - Move the super handler to `swc_ecma_transforms_base`.
 - Create a new own crate for the super handler.

The second one seems like a correct approach.
